src/lang/db/semantic.rs

```diff
@@ -68,7 +98,21 @@ pub trait LsSemanticGroup: Upcast<dyn SemanticGroup> + SemanticGroup {
     /// return a [`ModuleFileId`] pointing to the main, user-written file of the module.
     fn find_module_file_containing_node(&self, node: &SyntaxNode) -> Option<ModuleFileId> {
         let module_id = self.find_module_containing_node(node)?;
-        let file_index = FileIndex(0);
+        let file = node.stable_ptr(self.upcast()).file_id(self.upcast());
+
+        let i = self
+            .module_files(module_id)
+            .map(|files| {
+                files
+                    .iter()
+                    .enumerate()
+                    .find(|(_, f)| **f == file)
+                    .map(|(i, _)| i)
+                    .unwrap_or_default()
+            })
+            .unwrap_or_default();
+
+        let file_index = FileIndex(i);
```

Comment:
The description did not change - i do not understand what you're doing here then